// survey-angular-ui/esm2015/questions/ranking-item-content.component.js
import { Component, Input } from "@angular/core";
import { AngularComponentFactory } from "../component-factory";
import { EmbeddedViewContentComponent } from "../embedded-view-content.component";
import * as i0 from "@angular/core";
import * as i1 from "../survey-string.component";
export class RankingItemContentComponent extends EmbeddedViewContentComponent {
}
RankingItemContentComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: RankingItemContentComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
RankingItemContentComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: RankingItemContentComponent, selector: "sv-ranking-item", inputs: { item: "item", cssClasses: "cssClasses" }, usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n  <div [class]=\"cssClasses.controlLabel\">\n    <sv-ng-string [model]=\"item.locText\"></sv-ng-string>\n  </div>\n</ng-template>\n", styles: [":host { display: none; }"], components: [{ type: i1.SurveyStringComponent, selector: "sv-ng-string, '[sv-ng-string]'", inputs: ["model"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: RankingItemContentComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ranking-item",
                    templateUrl: "./ranking-item-content.component.html",
                    styles: [":host { display: none; }"]
                }]
        }], propDecorators: { item: [{
                type: Input
            }], cssClasses: [{
                type: Input
            }] } });
AngularComponentFactory.Instance.registerComponent("sv-ranking-item", RankingItemContentComponent);
