// survey-angular-ui/esm2015/components/loading-indicator/loading-indicator.component.js
import { Component } from "@angular/core";
import { EmbeddedViewContentComponent } from "../../embedded-view-content.component";
import * as i0 from "@angular/core";
import * as i1 from "../svg-icon/svg-icon.component";
export class LoadingIndicatorComponent extends EmbeddedViewContentComponent {
}
LoadingIndicatorComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: LoadingIndicatorComponent, deps: null, target: i0.ɵɵFactoryTarget.Component });
LoadingIndicatorComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: LoadingIndicatorComponent, selector: "sv-ng-loading-indicator", usesInheritance: true, ngImport: i0, template: "<ng-template #template>\n    <div class=\"sd-loading-indicator\">\n        <svg [iconName]=\"'icon-loading'\" [size]=\"'auto'\" sv-ng-svg-icon></svg>\n    </div>\n</ng-template>", styles: [":host{display:none}\n"], components: [{ type: i1.SvgIconComponent, selector: "'[sv-ng-svg-icon]'", inputs: ["size", "width", "height", "iconName", "partCss", "css", "title"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: LoadingIndicatorComponent, decorators: [{
            type: Component,
            args: [{
                    templateUrl: "./loading-indicator.component.html",
                    selector: "sv-ng-loading-indicator",
                    styleUrls: ["../../hide-host.scss"]
                }]
        }] });
