// survey-angular-ui/esm2015/components/popup/popup.component.js
import { Component, Input, ViewChild } from "@angular/core";
import { BaseAngular } from "../../base-angular";
import { createPopupViewModel } from "survey-core";
import * as i0 from "@angular/core";
import * as i1 from "./popup-container.component";
export class PopupComponent extends BaseAngular {
    constructor(viewContainerRef, changeDetectorRef) {
        super(changeDetectorRef, viewContainerRef);
    }
    getModel() {
        return this.popupModel;
    }
    onModelChanged() {
        var _a;
        if (this.model) {
            this.model.resetComponentElement();
            this.model.dispose();
        }
        this.model = createPopupViewModel(this.popupModel, (_a = this.viewContainerRef) === null || _a === void 0 ? void 0 : _a.element.nativeElement);
        this.setContainerElement();
    }
    setContainerElement() {
        var _a, _b;
        if (!!((_a = this.containerRef) === null || _a === void 0 ? void 0 : _a.nativeElement)) {
            const container = this.containerRef.nativeElement;
            this.model.setComponentElement(container, this.getTarget ? this.getTarget(container.parentElement) : (_b = container === null || container === void 0 ? void 0 : container.parentElement) === null || _b === void 0 ? void 0 : _b.parentElement, this.getArea ? this.getArea(container.parentElement) : undefined);
        }
    }
    ngAfterViewInit() {
        this.setContainerElement();
    }
    ngOnInit() {
        this.onModelChanged();
    }
    ngOnDestroy() {
        this.model.resetComponentElement();
        this.model.dispose();
        super.ngOnDestroy();
    }
}
PopupComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PopupComponent, deps: [{ token: i0.ViewContainerRef }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Component });
PopupComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: PopupComponent, selector: "sv-ng-popup, '[sv-ng-popup]'", inputs: { popupModel: "popupModel", getTarget: "getTarget", getArea: "getArea" }, viewQueries: [{ propertyName: "containerRef", first: true, predicate: ["containerRef"], descendants: true }], usesInheritance: true, ngImport: i0, template: "<div #containerRef><sv-ng-popup-container [model]='model' ></sv-ng-popup-container></div>", isInline: true, components: [{ type: i1.PopupBaseContainerComponent, selector: "sv-ng-popup-container, '[sv-ng-popup-container]'", inputs: ["model"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: PopupComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "sv-ng-popup, '[sv-ng-popup]'",
                    template: "<div #containerRef><sv-ng-popup-container [model]='model' ></sv-ng-popup-container></div>"
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { popupModel: [{
                type: Input
            }], getTarget: [{
                type: Input
            }], getArea: [{
                type: Input
            }], containerRef: [{
                type: ViewChild,
                args: ["containerRef"]
            }] } });
