// survey-angular-ui/esm2015/components/element-header/element-header.component.js
import { Component, HostBinding, HostListener, Input } from "@angular/core";
import * as i0 from "@angular/core";
import * as i1 from "../element-title/element-title.component";
import * as i2 from "../../survey-string.component";
import * as i3 from "../action-bar/action-bar.component";
import * as i4 from "@angular/common";
export class ElementHeaderComponent {
    constructor() {
    }
    get rootClass() {
        return this.element.cssHeader;
    }
    get width() {
        return this.element.titleWidth;
    }
    click(e) {
        if (this.element.clickTitleFunction !== undefined) {
            this.element.clickTitleFunction(e);
        }
    }
}
ElementHeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ElementHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ElementHeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.17", type: ElementHeaderComponent, selector: "'[sv-ng-element-header]'", inputs: { element: "element" }, host: { listeners: { "click": "click($event)" }, properties: { "class": "this.rootClass", "style.width": "this.width" } }, ngImport: i0, template: "<sv-ng-element-title [element]=\"element\"></sv-ng-element-title>\n<div *ngIf=\"element.hasDescriptionUnderTitle\" [class]=\"element.cssDescription\" [model]=\"element.locDescription\" [attr.id]=\"element.ariaDescriptionId\" [style.display]=\"element.hasDescription ? '': 'none' \" sv-ng-string></div>\n<ng-container *ngIf=\"element.hasAdditionalTitleToolbar\">\n<sv-ng-action-bar [model]=\"element.additionalTitleToolbar\"></sv-ng-action-bar>\n</ng-container>", components: [{ type: i1.ElementTitleComponent, selector: "sv-ng-element-title", inputs: ["element"] }, { type: i2.SurveyStringComponent, selector: "sv-ng-string, '[sv-ng-string]'", inputs: ["model"] }, { type: i3.ActionBarComponent, selector: "sv-action-bar, sv-ng-action-bar", inputs: ["model", "handleClick"] }], directives: [{ type: i4.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.17", ngImport: i0, type: ElementHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: "'[sv-ng-element-header]'",
                    templateUrl: "./element-header.component.html"
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { element: [{
                type: Input
            }], rootClass: [{
                type: HostBinding,
                args: ["class"]
            }], width: [{
                type: HostBinding,
                args: ["style.width"]
            }], click: [{
                type: HostListener,
                args: ["click", ["$event"]]
            }] } });
